File: 알고리즘/최이삭/Network/Network.js
function dfs(a, c, v, n) {
  if(c[v] === true) return;
  else {
      c[v] = true;
      for(let i = 0; i < n; i++) {
          if(a[v][i] === 1) {
              dfs(a, c, i, n);      
          }
      }
  }
}

function solution(n, computers) {
  let answer = 0;
  let check = [];
  for(let i = 0; i < n; i++) {
      check.push(false)
  }
  
  for(let i = 0; i < n; i++) {
      if(check[i] === false) {
          dfs(computers, check, i, n);
          answer++;
      }
  }
  return answer;
}

File: 알고리즘/최이삭/Network/Network.py
def dfs(computers, v, i, n):
    if(v[i] == True):
        return
    else:
        v[i] = True
        for j in range(0, n):
            if(computers[i][j] == 1):
              dfs(computers, v, j, n)
    

def solution(n, computers):
    visit = {}
    for i in range(0, n):
        visit[i] = False
    answer = 0
    for i in range(0, n):
        if(visit[i] == False):
            dfs(computers, visit, i, n)
            answer += 1
    return answer